### DISCUSSION TITLE
No native build was found for platform=linux arch=x64 runtime=node abi=64 uv=1 libc=glibc

### DISCUSSION THREAD

**Discussion by pbteja1998** [vercel/vercel#4581] [Help] [answered]
I am getting the following error when my application is deployed.

```
{
  "errorType": "Error",
  "errorMessage": "No native build was found for platform=linux arch=x64 runtime=node abi=64 uv=1 libc=glibc",
  "stack": [
    "Error: No native build was found for platform=linux arch=x64 runtime=node abi=64 uv=1 libc=glibc",
    "    at Function.load.path (/var/task/node_modules/node-gyp-build/index.js:56:9)",
    "    at load (/var/task/node_modules/node-gyp-build/index.js:20:30)",
    "    at Object.<anonymous> (/var/task/node_modules/leveldown/binding.js:1:43)",
    "    at Module._compile (internal/modules/cjs/loader.js:778:30)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)",
    "    at Module.load (internal/modules/cjs/loader.js:653:32)",
    "    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)",
    "    at Function.Module._load (internal/modules/cjs/loader.js:585:3)",
    "    at Module.require (internal/modules/cjs/loader.js:692:17)",
    "    at require (internal/modules/cjs/helpers.js:25:18)",
    "    at Object.<anonymous> (/var/task/node_modules/leveldown/leveldown.js:3:17)",
    "    at Module._compile (internal/modules/cjs/loader.js:778:30)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)",
    "    at Module.load (internal/modules/cjs/loader.js:653:32)",
    "    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)",
    "    at Function.Module._load (internal/modules/cjs/loader.js:585:3)",
    "    at Module.require (internal/modules/cjs/loader.js:692:17)",
    "    at require (internal/modules/cjs/helpers.js:25:18)",
    "    at Object.<anonymous> (/var/task/node_modules/level/level.js:1:44)",
    "    at Module._compile (internal/modules/cjs/loader.js:778:30)",
    "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)",
    "    at Module.load (internal/modules/cjs/loader.js:653:32)"
  ]
}
```
The application works locally with `vercel dev`. So, I am assuming it has something to do with the environment that the app is deployed at. 

If someone faced similar problem, please help.

Tell me if you need any more information. ( I am using node `10.x`)

**Comment by styfle** [comment, marked as answer]
New deployments default to the latest Node LTS version which is `12.x`.

You can select an older version by defining the `engines` property in your `package.json` file.

```json
{
  "engines": {
    "node": "10.x"
  }
}
```

https://vercel.com/docs/runtimes#official-runtimes/node-js/node-js-version

That being said, I think a better solution is to upgrade your dependencies which will probably add support for Node 12.

**Reply by pbteja1998** [reply]
I already have node 10.x in package.json

**Reply by pbteja1998** [reply]
https://github.com/pbteja1998/MyBit-Go.app/blob/fix-issues/package.json

**Reply by pbteja1998** [reply]
@styfle any another reason why this error is occurring?

**Reply by styfle** [reply]
I see now. The `abi=64` means that Node 10 was selected correctly (this is the `NODE_MODULE_VERSION`).

https://nodejs.org/en/download/releases/

That means your dependency doesn't support Node 10.

**Reply by pbteja1998** [reply]
It's working fine locally with 10.x and I think the problem is with glibc library rather than node.

I searched through similar issues and they all seem to say that problem is with gcc/g++ version in vercel

**Reply by pbteja1998** [reply]
Even vercel dev works fine locally

**Reply by pbteja1998** [reply]
@styfle is there anything else that I can do to get around this error while keeping the node version to 10.x?

**Reply by pbteja1998** [reply]
I found the following thread which addresses similar issue. But I don't see any solution on how to fix it.

https://spectrum.chat/zeit/now/no-native-build-was-found-for-libc-glibc~d8e97b56-e6de-4f64-891e-663de5287cf8

https://github.com/vercel/vercel/issues/2866

**Reply by djfm** [reply]
if you're bundling with webpack, try adding:

```
node: {
  __dirname: true,
}
```

to your webpack config, that fixed it for me (albeit in a very different context, your mileage may vary)

**Reply by patsonkaushik** [reply]
same issue.

**Comment by jaredatron** [comment]
I'm running into the same issue when using `npm link` to a module that has `sodium-native` as a dependency.

**Comment by Elijer** [comment]
I was running parcel via npx and fixed this by deleting the npx folder in the .npm/ folder in my user directory.
